lib/bottles.rb

```diff
@@ -9,28 +12,23 @@ def verses(starting, ending)
   end
 
   def verse(number)
+    vessel = Vessel.new("bottle", "beer")
+    inventory = Inventory.new(vessel, number)
+
     case number
     when 0
-      "No more bottles of beer on the wall, " +
-      "no more bottles of beer.\n" +
-      "Go to the store and buy some more, " +
-      "99 bottles of beer on the wall.\n"
-    when 1
-      "1 bottle of beer on the wall, " +
-      "1 bottle of beer.\n" +
-      "Take it down and pass it around, " +
-      "no more bottles of beer on the wall.\n"
-    when 2
-      "2 bottles of beer on the wall, " +
-      "2 bottles of beer.\n" +
-      "Take one down and pass it around, " +
-      "1 bottle of beer on the wall.\n"
+      command = "Go to the store and buy some more"
+      next_inventory = Inventory.new(vessel, 99)
```

Comment:
Yes, if you could modify the private data of `Vessel` (or `Inventory`), you could introduce that problem.  But in fact you can't do that; instances of these classes are immutable. (Ok, instances in ruby are never really immutable because you can always use `instance_variable_set`, but that's cheating.

With immutable objects you always end creating a lot of them, but these classes are so trivial to construct that there's no real performance problem, and in exchange you get to keep the higher-level code very simple.